src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,125 +1,178 @@
 import java.time.LocalDateTime;
 import java.time.LocalTime;
 import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.Locale;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 public class JavaDateTimeApi {
+
+    public static final int DEFAULT_LENGTH_LOCAL_DATE = 10;
+    public static final String DELIMITER = "-";
+    public static final String SHORT_IDS_ECT = "ECT";
+    public static final String PATTERN_DATE_YYYY_MM_DD = "yyyyMMdd";
+    public static final String PATTERN_DATE_DD_MMM_YYY = "dd MMM yyy";
+    public static final String PATTERN_DATE_DD_MMMM_YYYY_HH_MM = "dd MMMM yyyy HH:mm";
+
     /**
      * Return the current date as a String depending on a query.
-     *
+     * <p>
      * The query can be passed for the whole date or for it's part:
-     *      - FULL - current date as a whole: year, month, day of month
-     *        formatted as `YYYY-MM-DD` (a default return value);
-     *      - YEAR - current year;
-     *      - MONTH - name of the current month;
-     *      - DAY - current day of month;
+     * - FULL - current date as a whole: year, month, day of month
+     * formatted as `YYYY-MM-DD` (a default return value);
+     * - YEAR - current year;
+     * - MONTH - name of the current month;
+     * - DAY - current day of month;
      * In any other case throw DateTimeException.
      **/
     public String todayDate(DateTimePart datePart) {
-        return "Today";
+        LocalDate localDate = LocalDate.now();
+        switch (datePart) {
+            case DAY:
+                return String.valueOf(localDate.getDayOfMonth());
+            case MONTH:
+                return String.valueOf(localDate.getMonth());
+            case YEAR:
+                return String.valueOf(localDate.getYear());
+            case FULL:
+                return String.valueOf(localDate);
+            default:
+                throw new DateTimeException("Don't valid dataPart " + datePart);
+        }
     }
 
     /**
      * Given an Array of 3 elements, where
-     *         - 1-st element is a `year`;
-     *         - 2-nd element is s `month`;
-     *         - 3-rd element is a `day of month`;
-     *
+     * - 1-st element is a `year`;
+     * - 2-nd element is s `month`;
+     * - 3-rd element is a `day of month`;
+     * <p>
      * Return Optional of a date built from these elements.
      */
     public Optional<LocalDate> getDate(Integer[] dateParams) {
-        return Optional.empty();
+        String collect = Arrays.stream(dateParams)
+                .map(String::valueOf)
+                .collect(Collectors.joining(DELIMITER));
+        return dateParams.length == 0 || collect.length() > DEFAULT_LENGTH_LOCAL_DATE
+                ? Optional.empty() : Optional.ofNullable(LocalDate.parse(collect));
```

Comment:
1) In both of these cases you may get empty optional(optional with null value); So I don't see them as the opposite.
2) Let's not check the specific case. You can get these parameters right from the array and catch exceptions that may be thrown

Reply:
done